src/Framework/PaymentGateways/Routes/GatewayRoute.php

```diff
@@ -2,74 +2,94 @@
 
 namespace Give\Framework\PaymentGateways\Routes;
 
-use Give\Framework\PaymentGateways\Contracts\OffsiteGatewayInterface;
 use Give\Framework\PaymentGateways\DataTransferObjects\GatewayRouteData;
 use Give\Framework\PaymentGateways\Exceptions\PaymentGatewayException;
+use Give\Framework\PaymentGateways\Log\PaymentGatewayLog;
 use Give\Framework\PaymentGateways\PaymentGateway;
 use Give\Framework\PaymentGateways\PaymentGatewayRegister;
+use Give\Framework\PaymentGateways\Traits\ResponseHelpers;
+
+use function Give\Framework\Http\Response\response;
 
 /**
  * @since 2.18.0
  */
 class GatewayRoute
 {
+    use ResponseHelpers;
+
     /**
-     * @since 2.18.0
+     * This is our entry point into the Gateway Routing system.
      *
      * @return void
      * @throws PaymentGatewayException
+     * @since 2.18.0
+     * @unreleased - validate secureRouteMethods
+     *
      */
     public function __invoke()
     {
         if ($this->isValidListener()) {
             /** @var PaymentGatewayRegister $paymentGatewaysRegister */
             $paymentGatewaysRegister = give(PaymentGatewayRegister::class);
+
+            // get all registered gateways
             $paymentGateways = $paymentGatewaysRegister->getPaymentGateways();
+
+            // get all registered gateway ids
             $gatewayIds = array_keys($paymentGateways);
 
+            // make sure required params are valid
             if (!$this->isValidRequest($gatewayIds)) {
                 throw new PaymentGatewayException('This route is not valid.');
             }
 
+            // create DTO from GET request
```

Comment:
@jonwaldstein We are sanitizing query param data. We can use `give_clean($_GET)` to pass sanitized data to `GatewayRouteData::fromRequest`

Reply:
👍  77038462